test/selenium_helpers.js

```diff
@@ -34,25 +31,163 @@ const setupSelenium = function(done) {
       })
     }
 
-  this.getLogs = () => {
-    return this.browser.manage().logs().get('browser')
+  browser.getLogs = function(){
+    return this.manage().logs().get('browser')
       .then(logs => console.log('Browser Logs:', logs))
   }
 
-  this.loginAs = loginAs
-  done()
+  browser.loginAs = function(githubId) {
+    return this.visit(`/__login/${githubId}`)
+    .then(session => {
+      const body = this.findElement(By.css('body'), 2000)
+      return this.wait(
+        until.elementTextContains(body, `logged in as ${githubId}`),
+        5000
+      )
+    })
+  }
+
+  browser.closeTabs = function(){
+    return this.getAllWindowHandles()
+      .then(handles => {
+        for(let i = handles.length -1; i > 0; i--){
+          this.switchTo().window(handles[i])
+          this.close()
+        }
+        this.switchTo().window(handles[0])
+      })
+  }
+
+  browser.checkNumberOfTabs = function() {
+    return this.getAllWindowHandles()
+      .then(handles => handles.length < 2)
+  }
+
+  browser.identifyTabPopup = function(timeout=2000) {
+    if(timeout <= 0) throw new Error('Out of time')
+    const _this = this
+    return this.checkNumberOfTabs()
+      .then(condition => {
+          if (condition) return Promise.resolve()
+          else{
+            setTimeout(function(){_this.identifyTabPopup(timeout-100)}, 100)
+          }
+      })
+  }
+
+  browser.shouldSee = function(text, timeout=2000){
+    const body = this.findElement(By.css('body'));
+    return this.wait(until.elementTextContains(body, text), timeout)
+      .catch(error => {
+        throw new Error(`expected page to contain text: ${JSON.stringify(text)}`)
+      })
+  }
+
+  browser.shouldSeeWithin = function(text, element, timeout=2000){
+    return this.wait(until.elementTextContains(this.findElement(element), text), timeout)
+      .catch(error => {
+        throw new Error(`expected ${element} to contain text: ${JSON.stringify(text)}`)
+      })
+  }
+
+  browser.shouldNotSeeWithin = function(text, element, timeout=2000){
+    return this.wait(until.elementTextContains(this.findElement(element), text), timeout)
+    .then(function(error) { if(error) throw new Error(`${text} exists on the page and therefore fails this test`)}, function (result) {
```

Comment:
I think we want something more like this

```js
browser.shouldNotSeeWithin = function(text, element, timeout=2000){
  return this.wait(until.elementTextContains(this.findElement(element), text), timeout)
    .catch(error => {
      return true
    })
    .then(() => {
      throw new Error(`expected NOT to see ${JSON.stringify(text)} within ${element}`)
    })
}
```

at the very least the formatting needs work but I'm also not sure this tests what we want to test